src/main/java/org/checkerframework/checker/crypto/CryptoAnnotatedTypeFactory.java

```diff
@@ -1,57 +1,87 @@
 import org.checkerframework.checker.crypto.qual.Bottom;
 import org.checkerframework.checker.crypto.qual.UnknownAlgorithmOrProvider;
 import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
 import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.ElementQualifierHierarchy;
 import org.checkerframework.framework.type.QualifierHierarchy;
-import org.checkerframework.framework.util.GraphQualifierHierarchy;
-import org.checkerframework.framework.util.MultiGraphQualifierHierarchy.MultiGraphFactory;
 import org.checkerframework.javacutil.AnnotationBuilder;
 import org.checkerframework.javacutil.AnnotationUtils;
 
 public class CryptoAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
 
+    /** The @{@link AllowedAlgorithms} annotation. */
+    protected final AnnotationMirror ALLOWEDALGORITHMS =
+            AnnotationBuilder.fromClass(elements, AllowedAlgorithms.class);
+    /** The @{@link AllowedProviders} annotation. */
+    protected final AnnotationMirror ALLOWPROVIDERS =
+            AnnotationBuilder.fromClass(elements, AllowedProviders.class);
+    /** The @{@link Bottom} annotation. */
+    protected final AnnotationMirror BOTTOM = AnnotationBuilder.fromClass(elements, Bottom.class);
+    /** The @{@link UnknownAlgorithmOrProvider} annotation. */
+    protected final AnnotationMirror UNKNOWNALGORITHMORPROVIDER =
+            AnnotationBuilder.fromClass(elements, UnknownAlgorithmOrProvider.class);
+
     public CryptoAnnotatedTypeFactory(BaseTypeChecker checker) {
         super(checker);
         this.postInit();
     }
 
     @Override
-    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
-        return new CryptoQualifierHierarchy(
-                factory, AnnotationBuilder.fromClass(elements, Bottom.class));
+    protected QualifierHierarchy createQualifierHierarchy() {
+        return new CryptoQualifierHierarchy(this.getSupportedTypeQualifiers(), elements);
     }
 
-    private static final class CryptoQualifierHierarchy extends GraphQualifierHierarchy {
+    private final class CryptoQualifierHierarchy extends ElementQualifierHierarchy {
 
-        CryptoQualifierHierarchy(MultiGraphFactory factory, AnnotationMirror bottom) {
-            super(factory, bottom);
+        /**
+         * Creates a CryptoQualifierHierarchy from the given classes.
+         *
+         * @param qualifierClasses classes of annotations that are the qualifiers for this hierarchy
+         * @param elements element utils
+         */
+        public CryptoQualifierHierarchy(
+                Collection<Class<? extends Annotation>> qualifierClasses, Elements elements) {
+            super(qualifierClasses, elements);
         }
 
         @Override
         public boolean isSubtype(final AnnotationMirror subtype, final AnnotationMirror supertype) {
-            if (AnnotationUtils.areSameByClass(supertype, UnknownAlgorithmOrProvider.class)
```

Comment:
I think you changed these because they are deprecated right? https://github.com/typetools/checker-framework/pull/4492/files